pkg/loadinfo/loadinfo.go

```diff
@@ -36,46 +38,89 @@ func toMB(total uint64) uint64 {
 	return total / 1024 / 1024
 }
 
+func toPercent(part uint64, total uint64) float64 {
+	return float64(part) / float64(total) * 100
+}
+
+func pids() ([]int, error) {
+	//scan /proc/*/exe to find all active processes
+	matches, err := filepath.Glob("/proc/[0-9]*/exe")
+	if err != nil {
+		return nil, err
+	}
+
+	pids := make([]int, len(matches))
+	for _, file := range matches {
+		//extract the pid from the path
+		pid := filepath.Base(filepath.Dir(file))
+		ipid, _ := strconv.Atoi(pid)
+		pids = append(pids, ipid)
+	}
+
+	return pids, nil
+}
+
+func cpuPercent(stat procfs.ProcStat) float64 {
+	starttime, err := stat.StartTime()
+	if err != nil {
+		return 0
+	}
+
+	// calculate the percentage of CPU used since the process started
+	return (stat.CPUTime() / (float64(time.Now().Unix()) - starttime) * 100)
+}
+
 // LogCurrentSystemLoad logs the current system load and lists all processes
 // consuming more than cpuWatermark of the CPU
 func LogCurrentSystemLoad(logFunc LogFunc) {
-	loadInfo, err := load.Avg()
+	loadInfo, err := loadavg.Get()
 	if err == nil {
 		logFunc("Load 1-min: %.2f 5-min: %.2f 15min: %.2f",
-			loadInfo.Load1, loadInfo.Load5, loadInfo.Load15)
+			loadInfo.Loadavg1, loadInfo.Loadavg5, loadInfo.Loadavg15)
 	}
 
-	memInfo, err := mem.VirtualMemory()
-	if err == nil && memInfo != nil {
+	memInfo, err := memory.Get()
+	if err == nil {
 		logFunc("Memory: Total: %d Used: %d (%.2f%%) Free: %d Buffers: %d Cached: %d",
-			toMB(memInfo.Total), toMB(memInfo.Used), memInfo.UsedPercent, toMB(memInfo.Free), toMB(memInfo.Buffers), toMB(memInfo.Cached))
-	}
+			toMB(memInfo.Total), toMB(memInfo.Used), toPercent(memInfo.Used, memInfo.Total), toMB(memInfo.Free), toMB(memInfo.Buffers), toMB(memInfo.Cached))
 
-	swapInfo, err := mem.SwapMemory()
-	if err == nil && swapInfo != nil {
 		logFunc("Swap: Total: %d Used: %d (%.2f%%) Free: %d",
-			toMB(swapInfo.Total), toMB(swapInfo.Used), swapInfo.UsedPercent, toMB(swapInfo.Free))
+			toMB(memInfo.SwapTotal), toMB(memInfo.SwapUsed), toPercent(memInfo.SwapUsed, memInfo.SwapTotal), toMB(memInfo.SwapFree))
 	}
 
-	procs, err := process.Processes()
+	pids, err := pids()
 	if err == nil {
```

Comment:
We may want to inverse the logic here and eliminate the indent?

Reply:
I wanted to stick with the original code structure as much as possible to make it a replacement, not re-write.